app/screens/post_options/components/options/base_option.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {StyleSheet} from 'react-native';
+
+import FormattedText from '@components/formatted_text';
+import MenuItem from '@components/menu_item';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    destructive: {
+        color: theme.dndIndicator,
+    },
+    label: {
+        color: theme.centerChannelColor,
+        ...typography('Body', 200),
+    },
+    iconContainerStyle: {
+        marginLeft: 0,
+    },
+}));
+
+type BaseOptionType = {
+    i18nId: string;
+    defaultMessage: string;
+    iconName: string;
+    onPress: () => void;
+    testID: string;
+    isDestructive?: boolean;
+}
+
+const BaseOption = ({
+    i18nId,
+    defaultMessage,
+    iconName,
+    onPress,
+    testID,
+    isDestructive = false,
+}: BaseOptionType) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const labelStyles = useMemo(() => {
+        if (isDestructive) {
+            return StyleSheet.flatten([styles.label, styles.destructive]);
+        }
+        return styles.label;
+    }, [theme, isDestructive, styles]);
```

Comment:
Do we need to depend on theme also? Style is not enough?

Reply:
Yes, style is enough.